src/components/pages/History/ComputeJobs/index.tsx

```diff
@@ -229,9 +230,22 @@ export default function ComputeJobs(): ReactElement {
       } finally {
         setIsLoading(false)
       }
+      return true
     }
+    let intervalId = intervalIdRef.current
+    if (data.tokenOrders.length === 0) {
+      intervalId && clearInterval(intervalId)
+      return
+    }
+    if (intervalId) return
     getJobs()
-  }, [ocean, account, data, config?.metadataCacheUri])
+    intervalId = window.setInterval(function () {
```

Comment:
React has its own `setInterval` which can be used here so it's just `setInterval` instead of `window. setInterval`. In general almost never a reason to access browser-only `window` object, which can be problematic in a React/Gatsby app. This then might also allow ditching the `useRef` usage but not sure, having hard time figuring out what exactly is going on in that mega-method `getJobs()`.

Guess what would help as a code improvement is to get all logic for the refetching into it's own `useEffect()`. But that would imply moving `getJobs()` out of it, either have it wrapped in `useCallback` or moved into helper method returning values instead of setting state. So that in the end in the `useEffect` we only control when fetching occurs, including the interval fetching. Could be refactor rabbit hole but could be worth it for maintenance